### DISCUSSION TITLE
[Next Image] Getting the sharp package for image optimization on production to work

### DISCUSSION THREAD

**Discussion by pilleck** [vercel/next.js#31336] [Help] [answered]
Hello everyone, 

right now I found out, that my production build of my next.js app is loading the images very slow. So I read the documentation and found that I should use the optional sharp package for image optimization. The problem is, I'm not sure if I get it to work correctly. 

The doc is pretty flat [here](https://nextjs.org/docs/messages/sharp-missing-in-production)...

Here are my questions: 

1. How can I check if the optional sharp package is not loaded?
2. When I set the `NEXT_SHARP_PATH=/tmp/node_modules/sharp` how can I check then if the package is loaded?
3. Instead of set the variable during the start script, can I set it in the .env file? 

Any help much appreciated!
Phil

**Comment by upteran** [comment, marked as answer]
Hi, if I understand correctly by code, image-optimizer module has check:

```
sharp = require(process.env.NEXT_SHARP_PATH || 'sharp')

// some code 

 if (sharp) {
  ..some "sharp" code logic
} else {
   console.warn( message from https://nextjs.org/docs/messages/sharp-missing-in-production)
}
```

If you not installed module (with set NEXT_SHARP_PATH or not, doesn't matter) you will see above warning, you can see this message in your app logs

**Reply by pilleck** [reply]
Hey @upteran, thank you for the message! 
Yes you're right! Right now I have it running and when there is no message `console.warn( message from https://nextjs.org/docs/messages/sharp-missing-in-production)` then it seems to work. I don't get this message anymore since I rollback my custom definition of the path to the sharp package. 

When I first tried it out, I checked my page (with around 20 images) and recognized that some images where not loaded quickly... so I thought `sharp` not running on my machine. Then I added the path to the sharp package and because I did this wrong, I run into different pitfalls... which lead me to create this discussion here.

Still remaining is the question, if it is the right path for a big website to use the image optimization by next/image and sharp, cause for me and this 20 images page, it seems to be too slow...

**Reply by tmaihoff** [reply]
Thanks for your post. I had the same problem. 

Thought I needed to specfy the path via env variable, but still getting the warning in the logs.

Just deleting the env variable and having the sharp package installed solved it
